Source/JavaScriptCore/wasm/WasmIPIntSlowPaths.cpp

```diff
@@ -73,14 +73,20 @@ namespace JSC { namespace IPInt {
 #define IPINT_CALLEE(callFrame) \
     static_cast<Wasm::IPIntCallee*>(callFrame->callee().asNativeCallee())
 
+#if USE(JSVALUE64)
+#define WASM_RETURN_JSVALUE(value) WASM_RETURN_TWO(std::bit_cast<void*>(value), 0)
+#elif USE(JSVALUE32_64)
+#define WASM_RETURN_JSVALUE(value) WASM_RETURN_TWO(std::bit_cast<void*>(JSValue::decode(value).tag()), std::bit_cast<void*>(JSValue::decode(value).payload()))
+#endif
+
 // For operation calls that may throw an exception, we return (0, <val>)
```

Comment:
Can we add a FIXME to have this match what JS does on 64-bit? i.e.` return (<exception>, 0)`

Reply:
Oh yes! You meant `return (exception, 1)` right, i.e. flip the order? That would indeed be neat! I'm not going to get to it anymore this week, but I might just do this next week because I noticed that its also inconsistent with how we deal with register pairs on JSVALUE32_64